src/ChainSafe.Gaming.Marketplace/MarketplaceConfig.cs

```diff
@@ -2,34 +2,26 @@
 // Copyright (c) PlaceholderCompany. All rights reserved.
 // </copyright>
 
+using System;
+
 namespace ChainSafe.Gaming.Marketplace
 {
-    using System.Reflection;
-    using ChainSafe.Gaming.Evm.Utils;
-
     /// <summary>
     /// Marketplace configuration.
     /// </summary>
-    public struct MarketplaceConfig : IMarketplaceConfig
+    [Serializable]
+    public class MarketplaceConfig : IMarketplaceConfig
     {
-        /// <summary>
-        /// Gets constructor instantiates the class.
-        /// </summary>
-        public static MarketplaceConfig Default => new MarketplaceConfig
-        {
-            EndpointOverride = "https://api.gaming.chainsafe.io",
-            MarketplaceContractAbi = ReadDefaultAbiFromResources(),
-        };
-
         /// <inheritdoc/>
         public string? EndpointOverride { get; set; }
 
         /// <inheritdoc/>
-        public string? MarketplaceContractAbi { get; set; }
+        public string MarketplaceContractAbi { get; set; }
```

Comment:
Could rename this into `MarketplaceContractAbiOverride` and add another readonly property that will either use the override when it's not null or `ReadDefaultAbiFromResources`. All of this can sit happily in `IMarketplaceConfig`.